src/ast/passes/semantic_analyser.cpp

```diff
@@ -449,11 +449,20 @@ void SemanticAnalyser::visit(Builtin &builtin)
       builtin.type.MarkCtxAccess();
       builtin.type.is_funcarg = true;
     }
+    else if (type == ProbeType::uprobe)
+    {
+      builtin.type = CreatePointer(CreateRecord("struct uprobe_args",
+                                                bpftrace_.structs.Lookup(
+                                                    "struct uprobe_args")),
```

Comment:
This lookup confused me for a bit. Maybe leave a comment along the lines of "Add a dummy struct type here. When the eventual FieldAccess is resolved it, it special cases the argument lookup".